src/routing_table/mod.rs

```diff
@@ -869,6 +890,26 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         Ok(())
     }
 
+    #[cfg(any(test, feature = "use-mock-crust"))]
+    pub fn our_name(&self) -> &T {
+        &self.our_name
+    }
+
+    /// Returns the list of contacts as a result of a merge to which we aren't currently connected,
+    /// but should be.
+    #[cfg(any(test, feature = "use-mock-crust"))]
+    pub fn needed(&self) -> &Groups<T> {
+        &self.needed
+    }
+
+    #[cfg(test)]
+    pub fn verify_invariant(&self) {
+        if let Err(_) = self.check_invariant() {
+            let message = format!("Invariant not satisfied for RT: {:?}", self);
+            panic!(message);
```

Comment:
I'd already noticed this and actually changed it locally to use a single `unwrap!()` statement to achieve this :)  However, the error isn't particularly helpful - it can only be `Error::InvariantViolation` and that doesn't specify _which_ invariant has been broken.  Even adding a `String` to the error variant would help a bit, or I could change the single variant into the six different cases we check for, or I could add a new error enum just for the invariant checks and nest that in `Error::InvariantViolation`?